src/service/parallax-impl.js

```diff
@@ -168,16 +109,24 @@ export class ParallaxElement {
 
     /** @private {number} */
     this.offset_ = 0;
+
+    /** @private {number} */
+    this.previousScroll_ = 0;
   }
 
   /**
    * Apply the parallax effect to the offset given how much the page
    * has moved since the last frame.
-   * @param {number} delta The movement of the base layer e.g. the page.
+   * @param {!./viewport-impl.Viewport} viewport.
    */
-  update(delta) {
+  update(viewport) {
```

Comment:
Why pass a reference to the full viewport object when you only use `getScrollTop`? Consider passing the value from the calling function.